rudder-core/src/main/scala/com/normation/rudder/policyMode/PolicyMode.scala

```diff
@@ -0,0 +1,83 @@
+/*
+*************************************************************************************
+* Copyright 2016 Normation SAS
+*************************************************************************************
+*
+* This file is part of Rudder.
+*
+* Rudder is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* In accordance with the terms of section 7 (7. Additional Terms.) of
+* the GNU General Public License version 3, the copyright holders add
+* the following Additional permissions:
+* Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+* Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+* Public License version 3, when you create a Related Module, this
+* Related Module is not considered as a part of the work and may be
+* distributed under the license agreement of your choice.
+* A "Related Module" means a set of sources files including their
+* documentation that, without modification of the Source Code, enables
+* supplementary functions or services in addition to those offered by
+* the Software.
+*
+* Rudder is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+*
+*************************************************************************************
+*/
+package com.normation.rudder.policyMode
+
+import net.liftweb.common.{Box,Full,Failure}
+
+sealed trait PolicyMode {
+  def name : String
+}
+
+case object Verify extends PolicyMode {
+  val name = "verify"
+}
+
+case object Enforce extends PolicyMode {
+  val name = "enforce"
+}
+
+object PolicyModeName {
+  val allModes : List[PolicyMode] = Verify :: Enforce :: Nil
+
+  def parse (value : String) : Box[PolicyMode] = {
+    allModes.find { _.name == value.toLowerCase() } match {
+      case None =>
+         Failure(s"Unable to parse policy mode name '${value}'. was expecting ${allModes.map(_.name).mkString("'", "' or '", "'")}.")
+      case Some(mode) =>
+        Full(mode)
+    }
+  }
+}
+
+case class GlobalPolicyMode (
+    mode : PolicyMode
+  , overridable : Boolean
+)
+
+object PolicyModeService {
+
+  def computeMode(globalValue : GlobalPolicyMode, nodeMode : Option[PolicyMode], directiveMode : Option[PolicyMode]) = {
+    if (globalValue.overridable) {
+      nodeMode match {
+        case Some(Verify) => Verify
+        case _ => directiveMode.getOrElse(globalValue.mode)
+      }
+    } else {
+      globalValue.mode
+    }
+  }
+}
```

Comment:
Could you add an other method that allows to combine several policyMode and get the result? It is necessary for now, if directives for a same technique have different modes. I believe we could just have a variant with that signature:
```
def computeMode(globalValue : GlobalPolicyMode, nodeMode : Option[PolicyMode], directivesMode : Seq[Option[PolicyMode]])
```

In fact, I'm wondering if we really want to have the one you did for now, because we always need to look for the whole technique's directives status applied to that node, not just one in isolation.